src/main/java/com/synopsys/integration/alert/channel/slack/descriptor/SlackUIConfig.java

```diff
@@ -27,27 +27,22 @@
 
 import org.springframework.stereotype.Component;
 
-import com.synopsys.integration.alert.channel.slack.SlackChannel;
 import com.synopsys.integration.alert.common.descriptor.config.field.ConfigField;
 import com.synopsys.integration.alert.common.descriptor.config.field.TextInputConfigField;
-import com.synopsys.integration.alert.common.descriptor.config.ui.DescriptorMetadata;
 import com.synopsys.integration.alert.common.descriptor.config.ui.UIConfig;
-import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
-import com.synopsys.integration.alert.common.enumeration.DescriptorType;
 
 @Component
 public class SlackUIConfig extends UIConfig {
 
-    @Override
-    public DescriptorMetadata generateDescriptorMetadata() {
-        return new DescriptorMetadata("Slack", "slack", SlackChannel.COMPONENT_NAME, DescriptorType.CHANNEL, ConfigContextEnum.DISTRIBUTION, "slack", setupFields());
+    public SlackUIConfig() {
+        super("Slack", "slack", "slack");
```

Comment:
Encapsulate lowercase slack strings into static final strings in the descriptor class since they are the URL and font-awesome icon name.